cadc-vos-server/src/main/java/org/opencadc/vospace/server/actions/CreateNodeAction.java

```diff
@@ -132,25 +133,58 @@ public void doAction() throws Exception {
-                throw ex;
+
+        // Extract here to remove later.
+        NodeProperty creatorJWT = clientNode.getProperty(VOS.PROPERTY_URI_CREATOR_JWT);
+        if (Utils.isAdmin(caller, nodePersistence)) {
+            if (creatorJWT != null && clientNode.ownerDisplay != null) {
+                // The "creator" property clashes with the "creator-jwt" property.
+                throw NodeFault.InvalidArgument.getStatus("BUG: " + VOS.PROPERTY_URI_CREATOR + " property already exists, but " + VOS.PROPERTY_URI_CREATOR_JWT
+                                                              + " is present. This should not happen.");
+            }
+
+            if (clientNode.ownerDisplay != null) {
+                // admin is allowed to assign a different owner
+                try {
+                    Subject tmp = new Subject();
+                    // HACK: known IdentityManager implementations prefer the HttpPrincipal aka
+                    // network username for toDisplayString(Subject) so that's the value that
+                    // normally gets put into node.ownerDisplay and rendered in node documents
+                    // so choosing HttpPrincipal here makes output and input (set owner) use the
+                    // same values... but seems kind of sketchy.
+                    tmp.getPrincipals().add(new HttpPrincipal(clientNode.ownerDisplay));
+                    clientNode.owner = im.augment(tmp);
+                } catch (Exception ex) {
+                    log.error("failed to map " + clientNode.ownerDisplay + " to a known user", ex);
+                    throw ex;
+                }
+            } else if (creatorJWT != null) {
+                log.debug("Creating user allocation on behalf of OIDC account.");
+                Subject nodeOwner = new Subject();
+                final String token = creatorJWT.getValue();
+                final AuthorizationTokenPrincipal authorizationTokenPrincipal = new AuthorizationTokenPrincipal(AuthenticationUtil.AUTHORIZATION_HEADER,
+                                                                                                                AuthenticationUtil.CHALLENGE_TYPE_BEARER
+                                                                                                                    + " " + token);
+
+                nodeOwner.getPrincipals().add(authorizationTokenPrincipal);
+
+                nodeOwner = AuthenticationUtil.validateSubject(nodeOwner);
+                nodeOwner = AuthenticationUtil.augmentSubject(nodeOwner);
+
+                log.debug("Setting new allocation node owner to " + im.toDisplayString(nodeOwner));
+                clientNode.owner = nodeOwner;
+                clientNode.ownerID = null; // just in case
+
+                log.debug("Creating user allocation on behalf of OIDC account: OK");
+            } else {
+                clientNode.owner = caller;
```

Comment:
This `clientNode.owner = caller` has to be the default as it covers normal "user create node"; it is now inside the `isAdmin` block. 

Maybe the best is to always assign it right before the `if (Utils.isAdmin(...) at line 136

Reply:
Good idea, thanks.